docs/sources/auth/generic-oauth.md

```diff
@@ -64,17 +66,21 @@ Grafana will attempt to determine the user's e-mail address by querying the OAut
 1. Query the `/emails` endpoint of the OAuth provider's API (configured with `api_url`) and check for the presence of an e-mail address marked as a primary address.
 1. If no e-mail address is found in steps (1-4), then the e-mail address of the user is set to the empty string.
 
-Grafana will also attempt to do role mapping through OAuth as described below.
+### Roles
+
+Grafana will check for the presence of a role using the [JMESPath](http://jmespath.org/examples.html) specified via the `role_attribute_path` configuration option. The JMESPath will be applied to the `id_token` first. If there is no match, the UserInfo endpoint specified via the `api_url` configuration option will be tried next. The result after evaluating the `role_attribute_path` JMESPath expression needs to be a valid Grafana role, i.e. `Viewer`, `Editor` or `Admin`.
```

Comment:
```suggestion
Grafana checks for the presence of a role using the [JMESPath](http://jmespath.org/examples.html) specified via the `role_attribute_path` configuration option. The JMESPath is applied to the `id_token` first. If there is no match, then the UserInfo endpoint specified via the `api_url` configuration option is tried next. The result after evaluation of the `role_attribute_path` JMESPath expression should be a valid Grafana role, for example, `Viewer`, `Editor` or `Admin`.
```

Reply:
Was following the tense from the existing documentation - don't want to change the tense for just this section by this PR, and think that changing the tense should be a separate PR.
Re: "needs to be" - this is from the previous text and is more accurate/correct.

Reply:
Don't think this should have been resolved and merged - "should" and "for example" make the documentation worse IMHO: makes it fuzzier and suggests the person setting up the integration has a choice, but it won't work if they don't set this correctly.